pkgs/applications/virtualization/xenserver-buildroot/default.nix

```diff
@@ -0,0 +1,54 @@
+{stdenv, fetchurl, fetchgit, nix-template-rpm }:
+
+# Xenserver buildroot of somehow around XenServer 6.5.
+# Xenserver is not an application by itself, but just a collection of components.
+# This package only provides the build inputs to actually build the components of xenserver.
+stdenv.mkDerivation rec {
+  name = "xenserver-${version}";
+  version = "2015012901";
+
+  src = fetchgit {
+    url = "https://github.com/xenserver/buildroot";
+    rev = "5e9892582c3733416bf8b6ebf9aece8cefa15b28";
+    sha256 = "6ee4288fb715e1315ff0cc7923ec1efe896be176d6697a7d56015b77b4317447";
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
+
+  buildInputs = [ nix-template-rpm ];
+
+    #cp ${ ./buildRootInput.nix } $out/usr/share/buildroot/buildRootInput.nix
+  installPhase = ''
+    mkdir -p $out/usr/share/buildroot
+    cp -r SPECS $out/usr/share/buildroot/
+    cp -r SOURCES $out/usr/share/buildroot/
+    cp ${ ./translation.table } $out/usr/share/buildroot/translation.table
+
+    #
+    # generate script to update buildRootInput.nix
+    #
+    mkdir -p $out/bin
+    cat << EOF > $out/bin/xapi-generate-nix-buildRootInput
+    #!$SHELL
+    if [ -e nixpkgs ] || [ -e buildRootInput.nix ] || [ -e translation.table ]
+    then
+      echo "Error: directory 'nixpkgs', file 'buildRootInput.nix' or file 'translation.table' does already exist!"
```

Comment:
"already exists" is more common phrasing than "does already exist". (The latter is easy to confuse for "doesn't exist", at least for me.)